packages/datadog-plugin-kafkajs/src/index.js

```diff
@@ -0,0 +1,95 @@
+'use strict'
+
+function createWrapProducer (tracer, config) {
+  return function wrapProducer (createProducer) {
+    return function producerWithTrace () {
+      const serviceName = config.service || `${tracer._service}-kafka`
+      const producer = createProducer.apply(this, arguments)
+
+      const send = producer.send
+
+      const tags = {
+        'service.name': serviceName,
+        'span.kind': 'producer',
+        'span.type': 'queue',
+        component: 'kafka'
+      }
+
+      producer.send = tracer.wrap('kafka.produce', { tags }, function (...args) {
+        const { topic, messages } = args[0]
+        const currentSpan = tracer.scope().active()
+
+        currentSpan.addTags({
+          'resource.name': topic,
+          'kafka.topic': topic,
+          'kafka.batch.size': messages.length
+        }
+        )
+        return send.apply(this, args)
+      })
+
+      return producer
+    }
+  }
+}
+
+function createWrapConsumer (tracer, config) {
+  return function wrapProcessEachMessage (Consumer) {
+    return function processEachMessageWithTrace () {
+      const serviceName = config.service || `${tracer._service}-kafka`
+      const consumer = Consumer.apply(this, arguments)
+      const run = consumer.run
+
+      const tags = {
+        'service.name': serviceName,
+        'span.kind': 'consumer',
+        'span.type': 'queue',
+        component: 'kafka'
+      }
+
+      consumer.run = async function ({ eachMessage, ...args }) {
```

Comment:
This function should not be async since it never awaits. It should also not be anonymous and be named something like `runWithTrace`.